RealmSwift/Tests/RealmTests.swift

```diff
@@ -979,4 +979,413 @@ class RealmTests: TestCase {
         realm.refresh()
         XCTAssertEqual(try! Realm().objects(SwiftBoolObject.self).count, 1)
     }
+
+// MARK: - Async Transactions
+#if REALM_ASYNC_WRITES
+
+    func testAsyncTransactionShouldWrite() {
+        let realm = try! Realm()
+        let asyncComplete = expectation(description: "async transaction complete")
+
+        try! realm.writeAsync { _ in
+            realm.create(SwiftStringObject.self, value: ["string"])
+        } _: {
+            let object = realm.objects(SwiftStringObject.self).first
+            XCTAssertEqual(object?.stringCol, "string")
+            asyncComplete.fulfill()
+        }
+
+        waitForExpectations(timeout: 1, handler: nil)
+    }
+
+    func testAsyncTransactionShouldWriteOnCommit() {
+        let realm = try! Realm()
+        let asyncComplete = expectation(description: "async transaction complete")
+        try! realm.beginAsyncWrite({ _ in
+            realm.create(SwiftStringObject.self, value: ["string"])
+
+            realm.commitAsyncWrite {
+                let object = realm.objects(SwiftStringObject.self).first
+                XCTAssertEqual(object?.stringCol, "string")
+                asyncComplete.fulfill()
+            }
+        })
+
+        waitForExpectations(timeout: 1, handler: nil)
+        XCTAssertEqual(realm.objects(SwiftStringObject.self).count, 1)
+    }
+
+    func testAsyncTransactionShouldCancel() {
+        let realm = try! Realm()
+        let asyncComplete = expectation(description: "async transaction complete")
+        asyncComplete.isInverted = true
+
+        let asyncTransactionId = try! realm.beginAsyncWrite({ _ in
+            realm.create(SwiftStringObject.self, value: ["string"])
+            realm.commitAsyncWrite {
+                asyncComplete.fulfill()
+            }
+        })
+
+        try! realm.cancelAsyncWrite(asyncTransactionId)
+
+        waitForExpectations(timeout: 1, handler: nil)
+        XCTAssertNil(realm.objects(SwiftStringObject.self).first)
+    }
+
+    func testAsyncTransactionShouldCancelWithoutCommit() {
```

Comment:
I'm not sure I understand this test.
What method call is cancelling the commit? What is causing the realm to still have no SwiftStringObjects?

Reply:
Oh nvm. There is no commit. 🤦